syntax = "proto3";

package user;

service UserService {
  rpc RegisterUser(UserRegistrationRequest) returns (UserResponse);
  rpc LoginUser(UserLoginRequest) returns (UserResponse);
  rpc GetUser(UserRequest) returns (UserResponse);
  rpc UpdateUser(UpdateUserRequest) returns (UserResponse);
}

message UserRegistrationRequest {
  string username = 1;
  string email = 2;
  string password = 3;
  string first_name = 4;
  string last_name = 5;
  string phone = 6;
  string address = 7;
}

message UserLoginRequest {
  string username = 1;
  string password = 2;
}

message UserRequest {
  string user_id = 1;
}

message UpdateUserRequest {
  string user_id = 1;
  string email = 2;
  string first_name = 3;
  string last_name = 4;
  string phone = 5;
  string address = 6;
}

message UserResponse {
  string user_id = 1;
  string username = 2;
  string email = 3;
  string first_name = 4;
  string last_name = 5;
  string phone = 6;
  string address = 7;
  string status = 8;
  string message = 9;
}
